crates/krun/src/launch.rs

```diff
@@ -22,6 +24,31 @@ pub enum LaunchResult {
     },
 }
 
+#[derive(Debug)]
+enum LaunchError {
+    Connection(std::io::Error),
+    Json(serde_json::Error),
+    Server(String),
+}
+
+impl Error for LaunchError {}
```

Comment:
Note to self:

> If there are no items, format the impl (including its `{}`) on a single line.

https://doc.rust-lang.org/nightly/style-guide/items.html#impls

The currently commented out rule we're using for `cargo +nightly fmt` is wrong:

https://github.com/slp/krun/blob/d0179f3c4ab0eaa07a8813a772db805bf5be9965/rustfmt.toml#L3